pkg/storage/db.go

```diff
@@ -151,10 +152,35 @@ func (db *DB) Init(ctx context.Context, postgresConfig *PostgresConfig) error {
 
 		}
 		initerr = db.RunScript(ctx, createScript)
+
+		if initerr == nil {
+			db.startPartitionGeneratorTimer(ctx)
+		}
 	})
 	return initerr
 }
 
+func (db *DB) startPartitionGeneratorTimer(ctx context.Context) {
+	// run every 12 hours, with the assumption that partition tables will never be
+	// this granular, thus partition tables are created well in advance of their need
+	ticker := time.NewTicker(12 * time.Hour)
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				if err := db.generatePartition(ctx); err != nil {
+					logger := domain.LoggerFromContext(ctx)
+					logger.Error(logs.StorageError{Reason: err.Error()})
```

Comment:
suggestion: error logs should be actionable. if a PartitionConflict error is returned, that would indicate we tried to create a partition, but one for the specified time already existed. We may want to log that type of error at an info level.